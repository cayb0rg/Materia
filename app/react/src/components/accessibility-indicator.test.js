/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import AccessibilityIndicator from './accessibility-indicator.jsx';

const getProps = (val = 'Full') => ({
	accessibility: {
		keyboard: val,
		screen_reader: val
	}
})

describe('AccessibilityIndicator', () => {
	it('should render', () => {
		const wrapper = shallow(<AccessibilityIndicator widget={getProps()} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	})

	it('should change component as props change expected', () => {
		// Full
		let component = shallow(<AccessibilityIndicator widget={getProps()} />);
		expect(component.find('#keyboard-access-level').text()).toBe('fully supported');
		expect(component.find('#screen-reader-access-level').text()).toBe('fully supported');

		// Limited
		component = shallow(<AccessibilityIndicator widget={getProps('Limited')} />);
		expect(component.find('#keyboard-access-level').text()).toBe('partially supported');
		expect(component.find('#screen-reader-access-level').text()).toBe('partially supported');

		// None
		component = shallow(<AccessibilityIndicator widget={getProps('None')} />);
		expect(component.find('#keyboard-access-level').text()).toBe('not supported');
		expect(component.find('#screen-reader-access-level').text()).toBe('not supported');

		// Random
		component = shallow(<AccessibilityIndicator widget={getProps('ab9lw2')} />);
		expect(component.find('#keyboard-access-level').text()).toBe('not supported');
		expect(component.find('#screen-reader-access-level').text()).toBe('not supported');
	});

	it('should render correctly when props not set', () => {
		const component = shallow(<AccessibilityIndicator />);
		expect(component.find('#keyboard-access-level').text()).toBe('not supported');
		expect(component.find('#screen-reader-access-level').text()).toBe('not supported');
	});
});
