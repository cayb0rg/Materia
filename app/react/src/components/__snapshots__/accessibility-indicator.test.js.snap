// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessibilityIndicator should render 1`] = `
<div
  className="feature-list accessibility-options"
>
  <div
    className="list-holder"
  >
    <span
      className="feature-heading"
    >
      Accessibility:
    </span>
    <ul>
      <li>
        <div
          className="icon-spacer"
        >
          <KeyboardIcon
            color="#5a5a5a"
          />
        </div>
        <span>
          Keyboard is 
          <span
            className="highlighted full"
            id="keyboard-access-level"
          >
            fully supported
          </span>
        </span>
      </li>
      <li>
        <div
          className="icon-spacer"
        >
          <ScreenReaderIcon
            color="#5a5a5a"
          />
        </div>
        <span>
          Screen reader is 
          <span
            className="highlighted full"
            id="screen-reader-access-level"
          >
            fully supported
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
